airflow/jobs.py

```diff
@@ -1925,20 +1920,31 @@ def _execute(self):
                         if key in started:
                             started.pop(key)
                         continue
-                    elif ti.state == State.FAILED:
-                        self.logger.error("Task instance {} failed".format(ti))
-                        failed.add(key)
-                        tasks_to_run.pop(key)
-                        if key in started:
-                            started.pop(key)
-                        continue
-                    elif ti.state == State.UPSTREAM_FAILED:
-                        self.logger.error("Task instance {} upstream failed".format(ti))
-                        failed.add(key)
-                        tasks_to_run.pop(key)
-                        if key in started:
-                            started.pop(key)
-                        continue
+                    # guard against externally modified tasks instances or
+                    # in case max concurrency has been reached at task runtime
+                    elif ti.state == State.NONE:
+                        self.logger.warning("FIXME: task instance {} state was set to "
+                                            "None externally. This should not happen")
+                        ti.set_state(State.SCHEDULED, session=session)
+
+                    if self.rerun_failed_tasks:
```

Comment:
Just want to clarify, if I start a backfill with the flag rerun_failed_tasks and some tasks failed during execution, we will set it to `State.SCHEDULED` so that it will be rerun again in the same backfill round?

Reply:
e.g  previously user tries to run a backfill for 10 days, each dag_run has 3 tasks, 9 dag_runs succeeded, one dag_run has 1 failed task.  Before this pr, backfill will encounter exception if the user tries to run the backfill again. Now with this flag, backfill will rerun those failed tasks.